tools/monitoring/mysql/database.go

```diff
@@ -49,36 +46,22 @@ func (e ErrorLog) String() string {
 		e.TimeStamp, e.Msg, e.JobName, e.PRNumber, e.BuildLogURL)
 }
 
-// PubsubMsgHandler adds record(s) to ErrorLogs table in database,
-// after parsing build log and compares the result with config yaml
-func PubsubMsgHandler(db *sql.DB, configURL, buildLogURL, jobname string, prNumber int) error {
-	config, err := config.ParseYaml(configURL)
-	if err != nil {
-		return err
-	}
-
-	errorLogs, err := ParseLog(buildLogURL, config.CollectErrorPatterns())
-	if err != nil {
-		return err
-	}
-
-	tx, err := db.Begin()
-	if err != nil {
-		return err
-	}
+// NewDB returns the DB object with an active database connection
+func NewDB(c *mysql.DBConfig) (*DB, error) {
+	db, err := c.Connect()
+	return &DB{db}, err
+}
 
-	stmt, err := tx.Prepare(logInsertStmt)
+// InsertErrorLog insert a new error to the ErrorLogs table
+func (db *DB) InsertErrorLog(errPat string, errMsg string, jobName string, prNum int, blogURL string) error {
+	stmt, err := db.Prepare(`INSERT INTO ErrorLogs(ErrorPattern, ErrorMsg, JobName, PRNumber, BuildLogURL, TimeStamp)
```

Comment:
Why don't use db.query instead of db.Prepare + stmt.Exec?

Reply:
[db.query](https://golang.org/pkg/database/sql/#DB.Query) is typically used in a select statements where it returns row as results. In an insert/update where there's a side effect, stmt.Exec returns Results which shows how many rows are modified instead. Since this SQL stmt doesn't return anything, using exec instead of query. 

https://stackoverflow.com/questions/50664648/why-even-use-db-exec-or-prepared-statements-in-golang